src/minitrace_grpc.rs

```diff
@@ -0,0 +1,106 @@
+use std::{net::{IpAddr, SocketAddr}, sync::mpsc::{Receiver, Sender}, task::{Context, Poll}};
+
+use minitrace::{FutureExt, Span};
+use minitrace_datadog::Reporter;
+use tonic::{body::BoxBody, codegen::http::HeaderValue, transport::Body};
+use tower::{Layer, Service};
+
+#[derive(Debug, Clone)]
+pub struct MinitraceGrpcMiddlewareLayer {
+    pub sender: Sender<MinitraceSpans>,
```

Comment:
Does this need to be public?

Reply:
Yeah, because I import/instantiate this in main.rs... unless I am missing another way to do that other than making it public in my rust-noob-ness 😄

Reply:
@piercetrey-figure It's cool, it's just a minor nit. Your code looks good otherwise. I mention it just as an issue of data hiding and being consistent with the context `MinitraceGrpcMiddlewareLayer` is instantiated in:
```
    Server::builder()
        .layer(MinitraceGrpcMiddlewareLayer { sender: datadog_sender })
        .add_service(health_service)
        .add_service(PublicKeyServiceServer::new(public_key_service))
        .add_service(MailboxServiceServer::new(mailbox_service))
```

You could create a `new` method for `MinitraceGrpcMiddlewareLayer` and create it like
`MinitraceGrpcMiddlewareLayer::new(sender)`.

Reply:
oh, I got you, also, I thought you were referring to the entire struct being public, not the field... Yeah, the constructor/new method makes sense, I'll add that in, thanks!